File: 프로그래머스_삼각 달팽이/두번째 시도_성공.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n) {
	int output[1001][1001] = { 0 };
	vector<int> answer;
	int col = 0, row = 0;
	int count = 1;
	int extent = 0;
	for (int i= 0;!output[col][row] && i<n; i += 3) {
		extent = n - i - 1;
		for (int j =0; j< extent; col++,j++) {
			output[col][row] = count++;
		}
		for (int j = 0; j < extent; row++,j++) {
			output[col][row] = count++;
		}
		for (int j = 0; j < extent; col--, row--,j++) {
			output[col][row] = count++;
		}
		if (!output[col][row])
			output[col][row] = count++;
		row ++;
		col += 2;
	}
	

	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
			if (output[i][j])
				answer.push_back(output[i][j]);
		}
	}
	return answer;
}

int main() {

	solution(4);
}

File: Programmers/프로그래머스_삼각 달팽이/두번째 시도_성공.py
def solution(n):
    answer = []
    output = [[0 for col in range(n+1)] for row in range(n+1)]
    col = 0
    row = 0
    count = 1
    one_num = 0
    cal = 0
    while True:
        cal = n - one_num - 1
        for i in range(0,cal):
            if col >n or row > n  or output[col][row] > 0 :break
            output[col][row] = count
            count+=1
            col+=1
            
        for i in range(0,cal):
            if col >n or row > n  or output[col][row] > 0 :break
            output[col][row] = count
            count+=1
            row+=1
        for i in range(0,cal):
            if col >n or row >n  or output[col][row] > 0 :break
            output[col][row] = count
            count += 1
            col-=1
            row-=1
        if(output[col][row] == 0):
            output[col][row] = count
            count += 1
        
        one_num += 3
        col += 2
        row += 1
        if col >n or row >n  or output[col][row] > 0 :break
    count = 0
    for i in range(n):
        for j in range(n):
            if output[i][j] > 0:
                answer.append(output[i][j])
    return answer
